### PR TITLE
Added Mattermost Notifications

### PR DESCRIPTION
**Pull request by anonaut**
this implements #611
i tested every defined notification and also the various options (override username/icon, custom channel)

### CODE REVIEW

File: Ombi.Api.Interfaces/IMattermostApi.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
+﻿#region Copyright
+// /************************************************************************
+//    Copyright (c) 2016 Jamie Rees
+//    File: IMattermostApi.cs
+//    Created By: Jamie Rees
```
Comment (tidusjar) [inline Ombi.Api.Interfaces/IMattermostApi.cs]: You have the same name as me? 😆

File: Ombi.Api.Models/Notifications/MattermostNotificationBody.cs [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    {
+        [JsonConstructor]
+        public MattermostNotificationBody()
+        {
+            username = "Ombi";
+        }
+
+        [JsonIgnore]
+        private string _username;
+        public string username
```
Comment (tidusjar) [inline Ombi.Api.Models/Notifications/MattermostNotificationBody.cs]: We could probably do something like this to make this slightly cleaner:

``` 
public string username { get; set; } = "Ombi";
```

**Comment by tidusjar** [comment]
Looks good!

**Comment by anonaut** [comment]
sorry, was pretty busy. i hope my first rebase worked as intended

**Comment by PotatoQuality** [comment]
@tidusjar  Can we merge this one?

**Comment by tidusjar** [comment]
Looks good! Thanks dude!

**Pull request merged**